torch/utils/_sympy/functions.py

```diff
@@ -202,7 +242,9 @@ def eval(cls, base, divisor):
                 return FloorDiv(base - term, divisor) + quotient
 
         try:
-            gcd = sympy.gcd(base, divisor)
+            gcd = simple_floordiv_gcd(base, divisor)
+            if equal_valued(gcd, 1) and isinstance(divisor, sympy.Add):
+                gcd = sympy.gcd(base, divisor)
```

Comment:
Oh! Is this one load bearing? I guess it is if the divisor got expanded.

Reply:
Yeah, unfortunately I ran into the `(x*y + x + y + 1) // (x + 1)` during internal testing and I don't want to invent a fancy new gcd algo lol. (In this case I could subtract and factorize, but general gcd probably would need repeated subtraction Euclid style.)

I also tried a version that uses `sympy.factor` but that loops on some inputs as well, so as bad as `sympy.gcd`.